### PR TITLE
Add PIP Attribute Finder sample

### PR DESCRIPTION
**Pull request by senthalan**
## Purpose
Add PIP Attribute Finder sample

### CODE REVIEW

File: entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPConstants.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.sample.entitlement.pip.attribute.finder;
+
+public class CustomPIPConstants {
+    public static final String SAMPLE_ATTRIBUTE_ID ="http://wso2.org/identity/sample/id";
```
Comment (vihanga-liyanage) [inline entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPConstants.java]: Shall we add a new line here?

File: entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPConstants.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public static final String SAMPLE_ATTRIBUTE_NAME ="http://wso2.org/identity/sample/name";
+    public static final String SAMPLE_CATEGORY ="http://wso2.org/identity/sample";
+
```
Comment (vihanga-liyanage) [inline entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPConstants.java]: Remove the unnecessary new line.

File: entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPAttributeFinder.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    @Override
+    public Set<String> getAttributeValues(String subject, String resource, String action, String environment,
+                                          String attributeId, String issuer) throws Exception {
+
+        throw new UnsupportedOperationException("Method unsupported in the context");
+    }
+
+    public void init(Properties properties) throws Exception {
+
+
```
Comment (vihanga-liyanage) [inline entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPAttributeFinder.java]: Remove the unnecessary new line.

File: entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPAttributeFinder.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            BagAttribute bagAttribute = (BagAttribute) context.getAttributeValue();
+            if (bagAttribute.size() > 0) {
+                sampleID = ((AttributeValue) bagAttribute.iterator().next()).encode();
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("Finding attributes for the context %1$s", sampleID));
+                }
+            }
+        }
+
+
```
Comment (vihanga-liyanage) [inline entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPAttributeFinder.java]: Remove the unnecessary new line.

File: entitlement/pip-sample/pom.xml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+    <properties>
+        <carbon.identity.framework.import.version.range>[5.7.0, 6.0.0)</carbon.identity.framework.import.version.range>
+        <carbon.identity.framework.version>5.11.148</carbon.identity.framework.version>
+        <balana.version>1.1.2</balana.version>
+        <carbon.kernel.version>4.4.27</carbon.kernel.version>
+        <commons-lang.wso2.version>2.6.0.wso2v1</commons-lang.wso2.version>
+        <balana.import.version.range>[1.1.0, 2.0.0)</balana.import.version.range>
+    </properties>
+</project>
```
Comment (vihanga-liyanage) [inline entitlement/pip-sample/pom.xml]: Add a new line at EOF.

File: entitlement/pip-sample/pom.xml [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                            org.wso2.balana.utils.policy.dto;version="${balana.import.version.range}",
+                            org.wso2.carbon.identity.entitlement;version="${carbon.identity.framework.import.version.range}",
+                            org.wso2.carbon.identity.entitlement.pip;version="${carbon.identity.framework.import.version.range}",
+                            org.wso2.carbon.identity.entitlement.common.dto;version="${carbon.identity.framework.import.version.range}",
+                            org.wso2.carbon.identity.entitlement.common.util;version="${carbon.identity.framework.import.version.range}",
+                            *;</Import-Package>
+                        <Export-Package>
+                            org.wso2.accenture.xacml.pip.*
+                        </Export-Package>
+                        <DynamicImports>*</DynamicImports>
```
Comment (vihanga-liyanage) [inline entitlement/pip-sample/pom.xml]: Is it necessary to use dynamic imports?

File: entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPConstants.java (hunk 1 above)
Comment (senthalan) [reply entitlement/pip-sample/src/main/java/org/wso2/sample/entitlement/pip/attribute/finder/CustomPIPConstants.java]: Added

**Pull request merged**